File: python/raindrops/raindrops.py
def raindrops(number):
    message = convert_to_message(number)
    return message if message else "{}".format(number)

def convert_to_message(number):
    return f"{'Pling' if is_three_a_factor(number) else ''}" \
           f"{'Plang' if is_five_a_factor(number) else ''}" \
           f"{'Plong' if is_seven_a_factor(number) else ''}"

def is_three_a_factor(number):
    return is_factor(number, 3)

def is_five_a_factor(number):
    return is_factor(number, 5)

def is_seven_a_factor(number):
    return is_factor(number, 7)

def is_factor(number, potentialFactor):
    return number % potentialFactor == 0

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert returns a string  generated based on the raindrops rules below.
// The rules of raindrops are that if a given number:
// - has 3 as a factor, add 'Pling' to the result.
// - has 5 as a factor, add 'Plang' to the result.
// - has 7 as a factor, add 'Plong' to the result.
// - does not have any of 3, 5, or 7 as a factor, the result should be the digits of the number.
func Convert(n int) string {
	if !isDivisibleByPlingPlangPlong(n) {
		return strconv.Itoa(n)
	}
	var result string
	if isPling(n) {
		result += "Pling"
	}
	if isPlang(n) {
		result += "Plang"
	}
	if isPlong(n) {
		result += "Plong"
	}
	return result
}

func isDivisibleByPlingPlangPlong(n int) bool {
	return isPling(n) || isPlang(n) || isPlong(n)
}

func isPling(n int) bool {
	return n%3 == 0
}

func isPlang(n int) bool {
	return n%5 == 0
}

func isPlong(n int) bool {
	return n%7 == 0
}
